packages/grafana-data/src/transformations/transformers/rowNumberToField.ts

```diff
@@ -0,0 +1,50 @@
+import { map } from 'rxjs/operators';
+
+import { DataFrame, Field, FieldType } from '../../types/dataFrame';
+import { DataTransformerInfo } from '../../types/transformations';
+import { ArrayVector } from '../../vector';
+
+import { DataTransformerID } from './ids';
+
+export interface RowNumberToFieldTransformerOptions {}
+
+export const rowNumberToFieldTransformer: DataTransformerInfo<RowNumberToFieldTransformerOptions> = {
+  id: DataTransformerID.rowNumberToField,
+  name: 'Row number to field',
+  description: 'Add the row number of the data frame as a field',
+  defaultOptions: {},
+
+  operator: (options) => (source) =>
+    source.pipe(
+      map((data) => {
+        if (!Array.isArray(data) || data.length === 0) {
+          return data;
+        }
+
+        return data.map((frame) => ({
+          ...frame,
+          fields: addRowNumberField(frame),
+        }));
+      })
+    ),
+};
+
+function addRowNumberField(frame: DataFrame): Field[] {
+  const rowNumberFieldIndex = frame.fields.findIndex((field) => field.name === 'row number');
```

Comment:
this part feels like hidden logic... is that important?

Reply:
also, internal field names with spaces?

Reply:
> this part feels like hidden logic... is that important?

It's there to make sure the row number field doesn't get duplicated. Otherwise you can get into a spiral where everytime transformations get triggered a new row number series gets created

> also, internal field names with spaces?

I'm not a regular contributor to this kind of thing. Is there a convention you prefer?

Reply:
Went with snake_case